### ISSUE TITLE
Coin selection algorithm for sending from CoinJoin account

### ISSUE THREAD

**Issue by hynek-jina** [trezor/trezor-suite#6584]
Spending from a CoinJoin account requires a special coin selection algorithm. The input to the algorithm is the requested amount to be spent:

1. Reduce the set of available UTXOs to the ones that:
- [x] satisfy the account’s anonymity threshold.
- [x] If the total value of this reduced set of UTXOs is less than the requested amount, then display "Insufficient annonymized funds".

Note: _It's not possible to send UTXOs during the critical phase. All operations on device are interrupted and UI is blocked by modal_ 
        
2. Choose a subset from the available UTXOs that satisfy the following criteria (ordered from highest to lowest priority):
- [x] If possible, always try to choose confirmed UTXOs (those that are in a mined transaction).
- [ ] If possible, always try to choose UTXOs that are not registered for a CoinJoin.
    - [ ] If it is possible only with registered UTXOs for CoinJoin, then deregister the UTXO that are selected.
- [ ] If possible, try to choose the UTXOs so that each chosen UTXO comes from a different previous transaction than the others. [We can skip this criterion for the November release. If it is complicated to implement, then we can do it later.]
- [ ] If possible, try to create a transaction with no change-outputs. Anything below the dust limit can be given to the miners. [We can skip this criterion for the November release. If it is complicated to implement, then we can do it later.]
- [ ] Minimize the number of UTXOs used.
- [ ] Ensure that change addresses for SignTx are never used for CoinJoin output addresses and vice versa.

3. In future versions we can offer an adjustment of the payment amount in UI, as seen below in Wasabi 2.0, which would offer the user to increase or decrease the payment amount to avoid change. This is out of scope for the MVP.

**Comment by hynek-jina** [comment]
Critical part is already implemented.. Therefore I'm lowering the priority to `medium`

**Comment by MiroslavProchazka** [comment]
Coinjoin feature was discontinued as of June 1st, 2024. I’m closing this issue
